enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -270,4 +287,32 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
       throw new EnforcerRuleException("Failed to collect dependency " + ex.getMessage(), ex);
     }
   }
+
+  private static String findPaths(DependencyNode root, Artifact artifact) {
+    ImmutableList.Builder<ImmutableList<DependencyNode>> result = ImmutableList.builder();
```

Comment:
This would be clearer if you added an overloaded findArtifact(root, artifact) method that didn't take the intermediate state arguments and returned the list.

Reply:
Added `ImmutableList<ImmutableList<DependencyNode>> findArtifactPaths(DependencyNode node, Artifact artifact)`